File: api-monitor-ui/src/app/services/dashboard.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { DashboardResponse } from '../models/dashboard-response';
import { ResponseTimePoint } from '../models/response-time-point';

@Injectable({
  providedIn: 'root'
})
export class DashboardService {

  private http = inject(HttpClient);

  private apiUrl = 'http://localhost:7070/api/dashboard';

  getDashboard(): Observable<DashboardResponse> {
    return this.http.get<DashboardResponse>(this.apiUrl);
  }

  getResponseTrend() {

  return this.http.get<ResponseTimePoint[]>(
    'http://localhost:7070/api/dashboard/response-trend'
  );
}
}

File: api-monitor/src/main/java/com/ananya/api_monitor/service/DashboardService.java
package com.ananya.api_monitor.service;

import com.ananya.api_monitor.dto.DashboardResponse;
import com.ananya.api_monitor.dto.ResponseTimePoint;

import java.util.List;

public interface DashboardService {

    DashboardResponse getDashboard();

    List<ResponseTimePoint> getResponseTrend();
}